vyper/codegen/core.py

```diff
@@ -1,26 +1,80 @@
 from vyper import ast as vy_ast
 from vyper.address_space import CALLDATA, DATA, IMMUTABLES, MEMORY, STORAGE
 from vyper.codegen.ir_node import Encoding, IRnode
-from vyper.codegen.types import (
-    DYNAMIC_ARRAY_OVERHEAD,
-    ArrayLike,
-    BaseType,
-    ByteArrayLike,
-    DArrayType,
-    EnumType,
-    MappingType,
-    SArrayType,
-    StructType,
-    TupleLike,
-    TupleType,
-    ceil32,
-    is_bytes_m_type,
-    is_decimal_type,
-    is_integer_type,
-)
 from vyper.evm.opcodes import version_check
 from vyper.exceptions import CompilerPanic, StructureException, TypeCheckFailure, TypeMismatch
-from vyper.utils import GAS_CALLDATACOPY_WORD, GAS_CODECOPY_WORD, GAS_IDENTITY, GAS_IDENTITYWORD
+from vyper.semantics.types import (
+    AddressT,
+    BoolT,
+    BytesM_T,
+    BytesT,
+    DArrayT,
+    DecimalT,
+    HashMapT,
+    IntegerT,
+    InterfaceT,
+    StructT,
+    TupleT,
+    _BytestringT,
+)
+from vyper.semantics.types.shortcuts import BYTES32_T, INT256_T, UINT256_T
+from vyper.semantics.types.subscriptable import SArrayT
+from vyper.semantics.types.user import EnumT
+from vyper.utils import (
+    GAS_CALLDATACOPY_WORD,
+    GAS_CODECOPY_WORD,
+    GAS_IDENTITY,
+    GAS_IDENTITYWORD,
+    ceil32,
+)
+
+DYNAMIC_ARRAY_OVERHEAD = 1
+
+
+def is_bytes_m_type(typ):
+    return isinstance(typ, BytesM_T)
+
+
+def is_numeric_type(typ):
+    return isinstance(typ, (IntegerT, DecimalT))
+
+
+def is_integer_type(typ):
+    return isinstance(typ, IntegerT)
+
+
+def is_decimal_type(typ):
+    return isinstance(typ, DecimalT)
+
+
+def is_enum_type(typ):
+    return isinstance(typ, EnumT)
```

Comment:
Would these make sense to inline for readability reasons?

Reply:
tried it both ways, but some sections of code just read more cleanly with the helper functions.